bot/exts/events/advent_of_code/_cog.py

```diff
@@ -185,15 +186,47 @@ async def join_leaderboard(self, ctx: commands.Context) -> None:
         brief="Get a snapshot of the PyDis private AoC leaderboard",
     )
     @whitelist_override(channels=AOC_WHITELIST_RESTRICTED)
-    async def aoc_leaderboard(self, ctx: commands.Context) -> None:
-        """Get the current top scorers of the Python Discord Leaderboard."""
+    async def aoc_leaderboard(
+            self,
+            ctx: commands.Context,
+            day_and_star: Optional[str] = None,
+            maximum_scorers: Optional[int] = 10
+    ) -> None:
+        """
+        Get the current top scorers of the Python Discord Leaderboard.
+
+        Additionally, you can provide a day-star(e.g. 2-1, as second day first star) to filter your query more.
+        """
+        if maximum_scorers > AocConfig.max_day_and_star_results or maximum_scorers <= 0:
+            raise commands.BadArgument(
+                "Unfortunately, you reached a limit. "
+                f"(The maximum number of results you can query is {AocConfig.max_day_and_star_results})"
+            )
         async with ctx.typing():
             try:
                 leaderboard = await _helpers.fetch_leaderboard()
-            except _helpers.FetchingLeaderboardFailed:
+            except _helpers.FetchingLeaderboardFailedError:
                 await ctx.send(":x: Unable to fetch leaderboard!")
                 return
-
+            if day_and_star:
+                # Fetches a dictionary that contains solvers in respect of day, and star.
+                # e.g. 1-1 will fetch the solvers of the first star of the first day and their completion time
+                per_day_and_star = json.loads(leaderboard['leaderboard_per_day_and_star'])
+                try:
+                    scorers = per_day_and_star[day_and_star][:maximum_scorers]
+                except KeyError:
+                    raise commands.BadArgument(
+                        "Day and star format has a notation of 'day-star' day being 1 to 25, and star is either 1 or "
+                        "2 (e.g. 3-1) "
+                    )
+                day, star = day_and_star.split("-")
+                codeblock = f"```csharp\nTop scorers for day {day}, star {star} in UTC timezone\n"
+                for scorer in scorers:
+                    time_data = datetime.fromtimestamp(scorer['completion_time']).strftime("%b.%d %I:%M %p")
+                    codeblock += f"[{scorer['member_name']}] {time_data}\n"
+                codeblock += "```"
+                await ctx.send(codeblock)
+                return
```

Comment:
It's a pandemic and these lines of code are not 6 feet apart!
```suggestion
                    )

                day, star = day_and_star.split("-")
                codeblock = f"```csharp\nTop scorers for day {day}, star {star} in UTC timezone\n"

                for scorer in scorers:
                    time_data = datetime.fromtimestamp(scorer['completion_time']).strftime("%b.%d %I:%M %p")
                    codeblock += f"[{scorer['member_name']}] {time_data}\n"

                codeblock += "```"
                await ctx.send(codeblock)
                return
```